addons/product/models/product_pricelist.py

```diff
@@ -292,36 +292,32 @@ def _get_partner_pricelist_multi(self, partner_ids):
         Partner = self.env['res.partner'].with_context(active_test=False)
         company_id = self.env.company.id
 
-        Property = self.env['ir.property'].with_company(company_id)
+        IrConfigParameter = self.env['ir.config_parameter'].sudo()
         Pricelist = self.env['product.pricelist']
         pl_domain = self._get_partner_pricelist_multi_search_domain_hook(company_id)
 
         # if no specific property, try to find a fitting pricelist
-        specific_properties = Property._get_multi(
-            'property_product_pricelist', Partner._name,
-            list(models.origin_ids(partner_ids)),  # Some NewID can be in the partner_ids
-        )
         result = {}
         remaining_partner_ids = []
-        for pid in partner_ids:
-            if (
-                specific_properties.get(pid)
-                and specific_properties[pid]._get_partner_pricelist_multi_filter_hook()
-            ):
-                result[pid] = specific_properties[pid]
-            elif (
-                isinstance(pid, models.NewId) and specific_properties.get(pid.origin)
-                and specific_properties[pid.origin]._get_partner_pricelist_multi_filter_hook()
-            ):
-                result[pid] = specific_properties[pid.origin]
+        for partner in Partner.with_company(company_id).browse(partner_ids):
+            if partner.specific_property_product_pricelist._get_partner_pricelist_multi_filter_hook():
+                result[partner.id] = partner.specific_property_product_pricelist
             else:
-                remaining_partner_ids.append(pid)
+                remaining_partner_ids.append(partner.id)
 
         if remaining_partner_ids:
             # get fallback pricelist when no pricelist for a given country
             pl_fallback = (
                 Pricelist.search(pl_domain + [('country_group_ids', '=', False)], limit=1) or
-                Property._get('property_product_pricelist', 'res.partner') or
+                # save data in ir.config_parameter instead of ir.default for
+                # res.partner.property_product_pricelist
+                # otherwise the data will become the default value while
+                # creating without specifying the property_product_pricelist
+                # however if the property_product_pricelist is not specified
+                # the result of the previous line should have high priority
+                # when computing
+                Pricelist.browse(int(IrConfigParameter.get_param(f'res.partner.property_product_pricelist_{company_id}'))) or
+                Pricelist.browse(int(IrConfigParameter.get_param('res.partner.property_product_pricelist'))) or
```

Comment:
I might have missed something, but I don't see where you set those config parameters :thinking:.
Or is it only to allow manual configuration when needed ?

Reply:
In this case
if we store a `product_pricelist` in ir.default for field `property_product_pricelist`, it will be treated as the default value of the non-company_dependent field.
when we create a `res.partner` without specifiying `property_product_pricelist`, ORM will use the `product_pricelist` as the default, which is equal to `record = self.create({'property_product_pricelist': product_pricelist})` and `record.property_product_pricelist == product_pricelist`

But in the above code
we ask `specific_property_product_pricelist` and `property_product_pricelist` don't have ir.defualt records
when create `res.partner` without specifiying `property_product_pricelist`, `record.property_product_pricelist` will be computed by the `_get_partner_pricelist_multi` which may or maynot be the `product_pricelist` saved in `ir.config.parameter`. So we can promise the business logic doesn't change.

In short
Logically the value to be saved here is neither a normal company dependent fallback nor a default value. It has a customized logic which shouldn't be stored in ir.defualt. Before this PR, `property_product_pricelist` was a non- company_dependent field but uses ir_property table in its own way. But now we cannot do the same thing.

Reply:
I don't really understand how it does answer my question.
Those two parameters are not set anywhere, and your comment in the code here is misleading or misplaced IMHO.

Reply:
Mhm, ok then. I'm not sure we want to keep the support for those two configurations, but it seems the safest way to migrate it :ok_hand:

Thanks for the explanation.